examples/ct/handlers.go

```diff
@@ -283,33 +283,54 @@ func addChainInternal(ctx context.Context, c LogContext, w http.ResponseWriter,
 		return http.StatusBadRequest, fmt.Errorf("failed to verify add-chain contents: %v", err)
 	}
 
-	// Build up the SCT and MerkleTreeLeaf. The SCT will be returned to the client and
-	// the leaf will become part of the data sent to the backend.
+	// This is now.
+	timestamp := uint64(c.TimeSource.Now().UnixNano() / millisPerNano)
+
+	// Build the MerkleTreeLeaf that gets sent to the backend, and make a trillian.LogLeaf for it.
 	var issuer *x509.Certificate
 	if len(chain) > 1 {
 		issuer = chain[1]
 	}
-	merkleLeaf, sct, err := signerFn(c.logKeyManager, chain[0], issuer, c.TimeSource.Now())
+	merkleLeaf, err := makeLeafFn(chain[0], issuer, timestamp)
 	if err != nil {
-		return http.StatusInternalServerError, fmt.Errorf("failed to build SCT and Merkle leaf: %v %v", sct, err)
+		return http.StatusBadRequest, fmt.Errorf("failed to build MerkleTreeLeaf: %v", err)
 	}
-
-	// Send the Merkle tree leaf on to the Log server.
 	leaf, err := buildLogLeafForAddChain(c, *merkleLeaf, chain)
 	if err != nil {
 		return http.StatusInternalServerError, fmt.Errorf("failed to build LogLeaf: %v", err)
 	}
+
+	// Send the Merkle tree leaf on to the Log server.
 	req := trillian.QueueLeavesRequest{LogId: c.logID, Leaves: []*trillian.LogLeaf{&leaf}}
```

Comment:
I don't think that's true; not much has changed, after all.

If this is about the Empty vs LogLeaf response, there are reasonable ways to do it (see previous comment).

If it's about batching, there are mechanisms for that *and* we can keep always keep the batch RPC if we like it (plus, [it's not like the CT personality creates >1 leaf at a time](https://github.com/google/trillian/blob/9e81d6b0a3b28db4a8a28f102ee694ae8de2392b/examples/ct/handlers.go#L302), at least not now).

If it's something else, let's talk about it. Being specific helps.

API compliance is about style, it shouldn't be a performance loss. It doesn't stand to reason that we'd force all of our APIs to be bad performance-wise. It's about having APIs that look and behave consistently and familiarly.

I'd appreciate it if we gave it a try; it's not so hard to change it now and it's bound to reap us benefits in the long run.